homeassistant/components/arve/entity.py

```diff
@@ -0,0 +1,49 @@
+"""Arve base entity."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from asyncarve import ArveSensProData
+
+from homeassistant.components.sensor import SensorEntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import ArveCoordinator
+
+
+@dataclass(frozen=True, kw_only=True)
+class ArveDeviceEntityDescription(SensorEntityDescription):
+    """Describes Arve device entity."""
+
+    value_fn: Callable[[ArveSensProData], float | int]
```

Comment:
you either inherit from the generic `EntityDescription` here, or you move this to `sensor.py` and type your description as `EntityDescription` below

Reply:
and I think I would prefer the latter